### ISSUE TITLE
Replace existing cookie-auditing UI with DevTools

### ISSUE THREAD

**Issue by pes10k** [brave/brave-browser#32190]
Currently you can learn about the cookies a site has set and has access to in Brave in two Brave-unique ways:

1. Address bar dropdown menu
<img width="326" alt="Screenshot 2023-08-10 at 18 23 49" src="https://github.com/brave/brave-browser/assets/76526/eac56b4a-90ed-4d6f-9acb-b0f7df0d49a7">
2. Cookies and site data option in settings UI (e.g., brave://settings/content/siteDetails?site=https%3A%2F%2Fgithub.com)
 <img width="706" alt="Screenshot 2023-08-10 at 18 24 27" src="https://github.com/brave/brave-browser/assets/76526/8d81f0e2-20c4-48e4-a210-4ebb77696748">

These are not ideal for several reasons:
- they only show cookies, when what users (likely) care about is all stored values (localStorage, Cache, etc)
- they are depricated upstream, so they may be heavy to carry soon
- they (partially) duplicate better, maintained UI that exists in devtools

This issue would replace the UI both of above described links point to, with instead opening up the devtools -> Application UI for the relevant site.

Note that one significant requirement here is that this *would not* open the relevant site (the way you usually need to with devtools); it would only open devtools itself (to prevent any interaction with the relevant site).

**Comment by sangwoo108** [comment]
It is my understanding that when we click "Cookies" in the pic 1 or "Cookies and Site and Data" in the pic 2, then 'devtools/application ui' should show up. 

Yeah, the flow for the first would work, but I think the latter wouldn't be possible as devtools are designed to be mapped to the acutal tab. So for the latter, we can't help but enhancing the existing settings page, but all my guess.

**Comment by pes10k** [comment]
> but I think the latter wouldn't be possible as devtools are designed to be mapped to the acutal tab

Yes, i think this would definitely require some change to devtools (though i dont know whether its a small or a big change)
